makePoplogCommander.sh

```diff
@@ -938,12 +938,14 @@ cat << \****
         deque_pop_front( argd );
         return processArgs( argd, base, flags, envv );
     } else {
-        fprintf( stderr, "Unexpected arguments:" );
-        for ( int i = 0; i < deque_length( argd ); i++ ) {
-            fprintf( stderr, " %s", (char *)deque_get( argd, i ) );
-        }
-        fprintf( stderr, "\n" );
-        return EXIT_FAILURE;
+        setUpEnvironment( base, flags, envv );
+        char * subpath = "/pop/getpoploglib/lib/getpoplog_run_subcommand.p";
+        char * path = safe_malloc( strlen( subpath ) + strlen( base ) + 1 );
+        char * p = stpcpy( path, base );
+        p = stpcpy( p, subpath );
+        deque_push_front( argd, path );
+        deque_push_front( argd, "pop11" );
+        execvp( "pop11", deque_as_array( argd ) );
     }
     perror( NULL );
     return EXIT_FAILURE;
```

Comment:
Doesn't this mean we'll always exit with a failed return status even when the subcommand was successful?

Reply:
The execvp will only return on a failure - otherwise the running process is replaced by the process running Poplog. I will add a comment.

Reply:
Really? I didn't get that understanding when reading the man page: https://man7.org/linux/man-pages/man3/perror.3.html
are you 100% sure?

```c
#include <errno.h>
#include <unistd.h>
char* argv[];
int main()  {
execvp( "blah", argv);
perror(NULL);
return 0;
}
```
running this returns 0.

Reply:
The perror should only be reached if the execvp fails to find "blah". From the online manual:

> The exec() functions return only if an error has occurred.  The return value is -1, and [errno](https://man7.org/linux/man-pages/man3/errno.3.html) is set to indicate the error.

There's no need to check the return value because it is guaranteed to be -1.